crates/turborepo-lib/src/rewrite_json.rs

```diff
@@ -0,0 +1,411 @@
+        output.push(':');
+    }
+    output.push_str(value);
+    for _ in 0..length {
+        output.push('}');
+    }
+
+    output
+}
+
+/**
+ * Given path segments, generate a JSON object member with an optional
+ * trailing separator.
+ */
+fn generate_member(path_segments: Vec<&str>, value: &str, separator: &str) -> String {
+    let mut output = String::new();
+    output.push('\"');
+    output.push_str(path_segments[0]);
+    output.push('\"');
+    output.push(':');
+    output.push_str(&generate_object(path_segments[1..].to_vec(), value));
+    output.push_str(separator);
+
+    output
+}
+
+/**
+ * Given a JSONC document and an object traversal path, `unset_path` will
+ * return a minimally-mutated JSONC document with all occurrences of the
+ * specified path removed.
+ */
+pub fn unset_path(json_document_string: &str, path: Vec<&str>) -> Result<String, RewriteError> {
+    let root = get_root(json_document_string)?;
+
+    // The key path can appear multiple times. This a vec that contains each time it
+    // occurs.
+    let path_ranges = find_all_paths(&root, &path, vec![]);
+
+    // We mutate this as we go.
+    let mut output: String = json_document_string.to_owned();
+
+    // We could either join overlapping ranges, or just carry it over.
+    // This elects to carry it over.
+    let mut last_start_position = None;
+
+    // We iterate in reverse since we're mutating the string.
+    path_ranges.iter().rev().for_each(|range| {
+        let end = match last_start_position {
+            Some(last_start_position) => {
+                if range.end > last_start_position {
+                    last_start_position
+                } else {
+                    range.end
+                }
+            }
+            None => range.end,
+        };
+
+        let is_overlapping = end != range.end;
+        let replacement_char = if is_overlapping {
+            ""
+        } else {
+            &range.replacement_char
+        };
+
+        output.replace_range(range.start..end, replacement_char);
+        last_start_position = Some(range.start);
+    });
+
+    Ok(output)
+}
+
+/**
+ * find_all_paths returns the list of ranges which define the specified
+ * token.
+ */
+fn find_all_paths<'a>(
+    current_node: &'a jsonc_parser::ast::Value<'a>,
+    target_path: &Vec<&'a str>,
+    current_path: Vec<&'a str>,
+) -> Vec<Range> {
+    let mut ranges: Vec<Range> = vec![];
+
+    // Early exit when it's impossible to have matching ranges.
+    if target_path.is_empty() {
+        return ranges;
+    }
+
+    // We can only find paths on objects.
+    if let jsonc_parser::ast::Value::Object(obj) = current_node {
+        // We need a reference to the previous and next property to identify if we're
+        // looking at the first or last node which need special handling.
+        let properties: Vec<_> = obj.properties.iter().collect();
+
+        for index in 0..properties.len() {
+            let property = properties[index];
```

Comment:
```suggestion
        for (index, property) in properties.iter().copied().enumerate() {
```

Reply:
I did a full refactor of this to function without indices and avoiding a `collect` call. What I really want is something that is `tuple_windows` but looks like this:

```rust
while let (previous: Option<&Thing>, current: &Thing, next: Option<&Thing>) = some_iterator.next() {
  let is_first = previous.is_none();
  let is_last = next.is_none();

  // Operate on `current` with context about previous and next.
}
```

`.peekable()` gives me half of this with a bit of ceremony, but nothing I could find offers `.previous()`.